Code/RDGeneral/testDict.cpp

```diff
@@ -12,13 +12,320 @@
 
 #include "types.h"
 #include <RDGeneral/Invariant.h>
+#include <RDGeneral/RDAny.h>
+#include <RDGeneral/Dict.h>
 #include <RDGeneral/RDLog.h>
 #include <RDGeneral/utils.h>
+#include <boost/shared_ptr.hpp>
+#include <vector>
 #include <ctime>
 
 using namespace RDKit;
 using namespace std;
 
+struct Foo {
+  int bar;
+  float baz;
+  ~Foo() { std::cerr << "deleted!" << std::endl; }
+};
+
+void testRDAny() {
+  std::cerr << "Testing RDValue" << std::endl;
+  {
+    RDAny v(-2147450880);
+  }
+  
+  {
+    int vi = 0;
+    RDValue v(0);
+    for (int i=0; i<100; ++i) {
+        vi += i;
+        v = rdvalue_cast<int>(v) + i;
+        CHECK_INVARIANT(vi == rdvalue_cast<int>(v), "Opps, bad variant");
```

Comment:
From a correctness point of view it doesn't matter, but all of these `CHECK_INVARIANT` uses can be replaced with `TEST_ASSERT`. It saves you the explanatory text at the end.